From 8bd1f493543d4403b154b3c52e89af1de919a3a6 Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <panto@antoniou-consulting.com>
Date: Mon, 22 Apr 2013 17:08:09 +0300
Subject: [PATCH 09/11] bone: Put priorities in built-in capes

Place priority entries to built-in cape entries. That way we can
make sure that the add one capes take precedence

Signed-off-by: Pantelis Antoniou <panto@antoniou-consulting.com>
---
 arch/arm/boot/dts/am335x-bone-common.dtsi | 44 ++++++++++++++++++-------------
 arch/arm/boot/dts/am33xx.dtsi             |  5 ++++
 2 files changed, 31 insertions(+), 18 deletions(-)

diff --git a/arch/arm/boot/dts/am335x-bone-common.dtsi b/arch/arm/boot/dts/am335x-bone-common.dtsi
index e341a08..327e592 100644
--- a/arch/arm/boot/dts/am335x-bone-common.dtsi
+++ b/arch/arm/boot/dts/am335x-bone-common.dtsi
@@ -25,7 +25,7 @@
 
 	am33xx_pinmux: pinmux@44e10800 {
 		pinctrl-names = "default";
-		pinctrl-0 = <&userleds_pins>;
+		pinctrl-0 = <&userled_pins>;
 
 		userled_pins: pinmux_userled_pins {
 			pinctrl-single,pins = <
@@ -51,11 +51,6 @@
 
 	ocp: ocp {
 
-		/* reset controller comes first (even if disabled) */
-		rstctl: rstctl@0 {
-			status = "disabled";
-		};
-
 		uart1: serial@44e09000 {
 			status = "okay";
 		};
@@ -138,10 +133,10 @@
 			 slot@4 {
 			 	ti,cape-override;
 			 	compatible = "ti,beaglebone-black";
-			 	board-name = "Bone-LT-eMMC-2G";
+			 	board-name = "BB-BONE-RST2";
 			 	version = "00A0";
 			 	manufacturer = "Texas Instruments";
-			 	part-number = "BB-BONE-EMMC-2G";
+			 	part-number = "BB-BONE-RST2";
 			 };
 
 			/* geiger cape version A0 without an EEPROM */
@@ -154,16 +149,6 @@
 				part-number = "BB-BONE-GEIGER";
 			};
 
-			/* Beaglebone black has it soldered on */
-			slot@6 {
-				ti,cape-override;
-				compatible = "ti,beaglebone-black";
-				board-name = "Bone-Black-HDMI";
-				version = "00A0";
-				manufacturer = "Texas Instruments";
-				part-number = "BB-BONELT-HDMI";
-			};
-
 			/* Nixie cape version A0 without an EEPROM */
 			slot@7 {
 				ti,cape-override;
@@ -221,6 +206,29 @@
 				manufacturer = "Adafruit";
 				part-number = "BB-BONE-RS232-01";
 			};
+
+			/* Beaglebone black has it soldered on */
+			 slot@100 {
+			 	ti,cape-override;
+				priority = <1>;
+			 	compatible = "ti,beaglebone-black";
+			 	board-name = "Bone-LT-eMMC-2G";
+			 	version = "00A0";
+			 	manufacturer = "Texas Instruments";
+			 	part-number = "BB-BONE-EMMC-2G";
+			 };
+
+			/* Beaglebone black has it soldered on */
+			slot@101 {
+				ti,cape-override;
+				priority = <1>;
+				compatible = "ti,beaglebone-black";
+				board-name = "Bone-Black-HDMI";
+				version = "00A0";
+				manufacturer = "Texas Instruments";
+				part-number = "BB-BONELT-HDMI";
+			};
+
 		};
 									  
 		/* mapping between board names and dtb objects */
diff --git a/arch/arm/boot/dts/am33xx.dtsi b/arch/arm/boot/dts/am33xx.dtsi
index 76fa03b..eb0d2d4 100644
--- a/arch/arm/boot/dts/am33xx.dtsi
+++ b/arch/arm/boot/dts/am33xx.dtsi
@@ -155,6 +155,11 @@
 			interrupts = <62>;
 		};
 
+		/* reset controller comes first (even if disabled) */
+		rstctl: rstctl@0 {
+			status = "disabled";
+		};
+
 		uart1: serial@44e09000 {
 			compatible = "ti,omap3-uart";
 			ti,hwmods = "uart1";
-- 
1.8.1.4

